test/manual/api/improved_api_tests.sh

```
@@ -0,0 +1,676 @@
+# FUNCTION: Update systems
+update_systems() {
+  echo "==== Updating Systems ===="
+  load_systems
+  
+  if [ -z "$CREATED_SYSTEM_IDS" ]; then
+    echo "No systems available. Run system creation first."
+    return 1
+  fi
+  
+  local update_count=0
+  local system_array=($CREATED_SYSTEM_IDS)
```

Shellcheck (0.10.0)

[warning] Quote to prevent word splitting/globbing, or split robustly with mapfile or read -a.

(SC2206)